app/resources.py

```diff
@@ -71,20 +75,81 @@ class BucketlistResource(Resource):
     @jwt_required()
     def get(self, id=None):
         bucketlist_fields = {
-                    'id': fields.Integer,
-                    'name': fields.String,
-                    'description': fields.String
-                }
+            'id': fields.Integer,
+            'name': fields.String,
+            'description': fields.String
+        }
 
         if id is not None:
-            bucketlist = Bucketlist.query.get(id)
+            bucketlist = Bucketlist.query.filter_by(
+                owner_id=current_identity['user_id'],
+                id=id).first()
             if bucketlist is not None:
                 return marshal(bucketlist, bucketlist_fields)
             else:
                 return {'message': 'requested id does not exist'}, 404
         else:
-            bucketlists = Bucketlist.query.all()
-            return marshal(bucketlists, bucketlist_fields)
+            if request.args:
+                parser = reqparse.RequestParser()
+                parser.add_argument('page',
+                                    type=int,
+                                    default=1,
+                                    location='args')
+                parser.add_argument('limit',
+                                    type=int,
+                                    default=10,
+                                    location='args')
+                parser.add_argument('q',
+                                    type=str,
+                                    location='args')
+                args = parser.parse_args(strict=True)
+
+                bucketlists = None
```

Comment:
Calling parse_args with strict=True ensures that an error is thrown if the request includes arguments your parser does not define